modules/ding_loan/plugins/content_types/loans.inc

```diff
@@ -178,15 +178,10 @@ function ding_loan_loans_form($form, &$form_state, $account, $loans = array()) {
 
       if (!is_object($entity) || empty($entity->reply)) {
         // Patron have this item in loans list, but opensearch can't find this
-        // record for current agencyId, so we load it separately.
-        // Here we are loading TingClientObject.
-        $ting_object = ting_get_object($loan->ding_entity_id);
-
-        // But we need TingEntity instance, so here we are populating this.
-        $ting_entity = new TingEntity();
-        $ting_entity->ding_entity_id = $entity->ding_entity_id;
-        $ting_entity->reply = $ting_object;
-        $entity = $ting_entity;
+        // record for current agencyId, so ask ding_provider for help:
+        $ding_entity_id = $loan->ding_entity_id;
```

Comment:
The `$loan` variable contents is a result of a successful call to the well performed on L60. So in case when an item is missing in the well the `$ding_entity_id` will be `NULL` with unknown consequences.